packages/core/src/StateNode.ts

```diff
@@ -1263,7 +1263,8 @@ class StateNode<
       configuration: resolvedConfiguration,
       transitions: stateTransition.transitions,
       children,
-      done: isDone
+      done: isDone,
+      tags: currentState?.tags
```

Comment:
this is kinda funky since we'll override this later on when we match some transition and this is outdated at the time of creating this `nextState` but it was an easy-to-do thing now - this happens in our internals, so hopefully this shouldn't be that observable.

We might call some expressions with that outdated info when processing the micro queue but that might be partially expected (this part already works in a different way than in SCXML since it updates the configuration as it goes and we don't do that).